src/components/LibraryDetailPage.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from "react";
+import { LibraryData } from "../interfaces";
+import { connect } from "react-redux";
+import ActionCreator from "../actions";
+import { Store } from "redux";
+import { State } from "../reducers/index";
+import LibraryDetailItem from "./LibraryDetailItem";
+import LibraryStageItem from "./LibraryStageItem";
+import SaveButton from "./SaveButton";
+
+export interface LibraryDetailPageStateProps {
+  fullLibrary?: LibraryData;
+}
+
+export interface LibraryDetailPageDispatchProps {
+  fetchData: () => Promise<LibraryData>;
+  editStages: (data: FormData) => Promise<void>;
+}
+
+export interface LibraryDetailPageOwnProps {
+  store?: Store<State>;
+  uuid: string;
+  toggle: (event: __React.MouseEvent) => void;
+}
+
+export interface LibraryDetailPageProps extends LibraryDetailPageStateProps, LibraryDetailPageDispatchProps, LibraryDetailPageOwnProps {}
+
+export class LibraryDetailPage extends React.Component<LibraryDetailPageProps, void> {
+
+  constructor(props: LibraryDetailPageProps) {
+    super(props);
+    this.submit = this.submit.bind(this);
+  }
+
+  componentWillMount() {
+    this.props.fetchData();
+  }
+
+  renderInfo() {
+    let fields = Object.keys(this.props.fullLibrary).map(label =>
+      <LibraryDetailItem key={label} label={label} value={this.props.fullLibrary[label]} />
+    );
+    return fields;
+  }
+
+  renderStages() {
+    let libraryStage = this.props.fullLibrary["library_stage"];
+    let registryStage = this.props.fullLibrary["registry_stage"];
+    return (
+      <form ref="form" className="well">
+        <input
+          type="hidden"
+          name="uuid"
+          value={this.props.uuid}
+        />
+        <LibraryStageItem key={"libraryStage"} label={"Library Stage"} value={libraryStage} />
+        <LibraryStageItem key={"registryStage"} label={"Registry Stage"} value={registryStage} />
+        <SaveButton
+          submit={this.submit}
+        />
+      </form>
+    );
+  }
+
+async submit(event: __React.MouseEvent): Promise<void> {
+    event.preventDefault();
+    let form = (this.refs["form"] as any);
+    const data = new (window as any).FormData(form);
+    await this.props.editStages(data);
+    this.props.toggle(event);
+    this.props.fetchData();
```

Comment:
This seems unnecessary because you are fetching data after submitting but before fetching the data you are closing the panel. When you close the panel you unmount the component so the new data won't be displayed anyway. You're relying on the top level `libraries` data for the stage level for the heading color so another reason why this doesn't seem necessary. When you open the panel and mount the component again, then you do fetch and render the latest data. But, this last part about mounting/unmounting also depends in `LibrariesListItem` where you check `this.props.active` to render the `ListDetailContainer`.

Reply:
Also, what you do decide to do here depends on refactoring so that the flash doesn't happen when a panel opens and closes.

Reply:
It is completely unnecessary; good catch!